src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -34,6 +32,95 @@ constexpr float default_queue_priority = 1.0f;
 
 namespace inexor::vulkan_renderer::wrapper {
 
+bool Device::device_extension_supported(const VkPhysicalDevice graphics_card, const std::string &extension_name) {
+    assert(graphics_card);
+    assert(!extension_name.empty());
+
+    std::uint32_t device_extension_count = 0;
+
+    // Query how many device extensions are available.
+    if (const auto result =
+            vkEnumerateDeviceExtensionProperties(graphics_card, nullptr, &device_extension_count, nullptr);
+        result != VK_SUCCESS) {
+        throw exceptions::VulkanException("Error: vkEnumerateDeviceExtensionProperties failed!", result);
+    }
+
+    std::vector<VkExtensionProperties> device_extensions(device_extension_count);
+
+    if (device_extension_count == 0) {
+        throw std::runtime_error("Error: No Vulkan device extensions available!");
+    } else {
+        // Store all available device extensions.
+        if (const auto result = vkEnumerateDeviceExtensionProperties(graphics_card, nullptr, &device_extension_count,
+                                                                     device_extensions.data());
+            result != VK_SUCCESS) {
+            throw exceptions::VulkanException("Error: vkEnumerateDeviceExtensionProperties failed!", result);
+        }
+    }
+
+    // Search for the requested device extension.
+    auto result = std::find_if(device_extensions.begin(), device_extensions.end(),
+                               [&](const VkExtensionProperties device_extension) {
+                                   return (strcmp(device_extension.extensionName, extension_name.c_str()) == 0);
+                               });
+
+    return result != device_extensions.end();
+}
+
+bool Device::device_layer_supported(const VkPhysicalDevice graphics_card, const std::string &layer_name) {
+    assert(graphics_card);
+    assert(!layer_name.empty());
+
+    std::uint32_t device_layer_count = 0;
+
+    // Query how many device layers are available.
+    if (const auto result = vkEnumerateDeviceLayerProperties(graphics_card, &device_layer_count, nullptr);
+        result != VK_SUCCESS) {
+        throw exceptions::VulkanException("Error: vkEnumerateDeviceLayerProperties failed!", result);
+    }
+
+    std::vector<VkLayerProperties> device_layers(device_layer_count);
+
+    if (device_layer_count == 0) {
+        throw std::runtime_error("Error: No Vulkan device extensions available!");
```

Comment:
No need for `else` after `throw`.

Reply:
I applied this to all other code blocks in device and instance wrapper.